### PR TITLE
Switch from Api 2 to Api 3.1

### PR DESCRIPTION
**Pull request by joshsouza**
https://github.com/russellcardullo/terraform-provider-pingdom/issues/42

This PR would fully drop support for the v2 api (which is deprecated soon), and thus this is not backwards compatible.
Also, I'm pretty new to golang, so please take care to verify that this is how you would want this code written. (I have some commented out code here still that probably needs to be deleted.)

Unfortunately I can't commit the time to making a forwards/backwards compatible version of this change, but our team has been working off of this branch as a dependency for our `terraform-provider-pingdom` for most of a year at this point, and it's been working great for us. 

Feel free to simply use this as a suggestion on how to take this code and make it forwards/backwards compatible.

### CODE REVIEW

**Comment by yurymuski** [comment]
@russellcardullo Hello, This PR is very usefull, please review it :)

**Comment by russellcardullo** [comment]
Thanks for the PR, will take a look next week.

**Comment by rpiskule-syn** [comment]
@russellcardullo 
I checked out @joshsouza 's PR, it looks pretty good. Pingdom is basically just supporting an API Token at this point (no need for a username and password).

For example:
```
echo "name=My+new+HTTP+check&type=http&host=www.google.com" > /tmp/body.txt
curl -X POST https://api.pingdom.com/api/3.1/checks -H "Authorization:Bearer $API_TOKEN" -d @/tmp/body.txt
```

This project uses your library as well, and basically they are stuck until you update your library. I was looking for a way to automate my pingdom checks either using your terraform module, or the ingress monitor, but both are broken until this gets merged.
https://github.com/stakater/IngressMonitorController/issues/214

BTW, thanks for all your hard work.

**Comment by joshsouza** [comment]
@russellcardullo Any chance you'll have some time to take a look at this in the next week? Just trying to set expectations on this end. We all appreciate your work here.

File: pingdom/pingdom.go [hunk 1]
```diff
@@ -70,36 +58,18 @@ func NewClientWithConfig(config ClientConfig) (*Client, error) {
...
-	config := ClientConfig{
-		User:         user,
-		Password:     password,
-		APIKey:       key,
-		AccountEmail: accountEmail,
-	}
-	c, _ := NewClientWithConfig(config)
-	return c
-}
+// func NewClient(key string) *Client {
```
Comment (russellcardullo) [inline pingdom/pingdom.go:66]: nit: what do you think about removing the commented out code here and just exclusively use `NewClientWithConfig`?

**Review by russellcardullo — APPROVED** [review]
@joshsouza thanks for the work on this PR and I apologize for the delay in reviewing.  Everything here looks great!

I was thinking about backward compatability and not sure how much the effort would be justified, especially given the 2.0/2.1 API will be EOL in December 2020.

I'm thinking of merging this to `master` and adding a new 2.X tag.  That will most certinaly break people who aren't pinning their dependencies, but they could switch back using the latest 1.X tag.

If there aren't any objections to breaking backward compatibility on master I'll merge this in a few days and update the tags.

Again, thanks for working on this!

File: pingdom/pingdom.go (hunk 1 above)
Comment (joshsouza) [reply pingdom/pingdom.go:66]: I can do that

**Comment by russellcardullo** [comment]
Thanks again for working on this @joshsouza.  I've just merged this and pushed a new v2.0.0 tag.  If anyone still needs to use the older Pingdom API they can use tag v1.1.0.

**Comment by rpiskule-syn** [comment]
@russellcardullo Thank you so much, we appreciate it!

**Comment by russellcardullo** [comment]
Ah I forgot about golangs v2 rule: https://golangnews.org/2020/09/golang-modules-have-v2-problem/

I'm going to tag the latest release as v1.2.0 in the meantime instead of v2.0.0.  That's not great from a semantic versioning perspective but the alternative is to change the package name to go-pingdom/v2.

**Comment by cce** [comment]
Hey, @joshsouza and @russellcardullo thanks for this effort! Some of the API that this PR removed from v2 ([Teams](https://docs.pingdom.com/api/#tag/Teams), [Users](https://docs.pingdom.com/api/#tag/Users), [Contact targets](https://docs.pingdom.com/api/#tag/Users/paths/~1users~1{userid}/post)) exists but with different JSON schemas in v3.1 (just [Teams](https://docs.pingdom.com/api/#tag/Teams) of [Contacts](https://docs.pingdom.com/api/#tag/Contacts)).

Given the close mapping between the REST API JSON and the Go API structs, it would be easier to add managing teams and contacts to the API in a backwards-incompatible way, than to make it backwards compatible.

Would you be interested in a backwards-incompatible PR for Teams & Contacts on the new v1.2 release track?

**Comment by joshsouza** [comment]
When I put in this PR initially, I was pretty sure they had removed team
and contact management, but also noticed that it's now available yesterday.
I've got a branch that reimplements that functionality which I spent all
yesterday on, and should be ready for some eyes today!

**Pull request merged**